src/extensionConflicts.ts

```diff
@@ -0,0 +1,68 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Red Hat, Inc. All rights reserved..
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { commands, Extension, extensions, window } from 'vscode';
+
+// A set of VSCode extension ID's that conflict with our extension
+const conflictingIDs = [
+  'haaaad.ansible',
+  'sysninja.vscode-ansible-mod',
+  'tomaciazek.ansible',
+  'vscoss.vscode-ansible',
+  'zbr.vscode-ansible',
+];
+
+// A set of VSCode extension ID's that are currently uninstalling
+const uninstallingIDs = new Set();
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function isExtensionPresent(obj: any): obj is Extension<any> {
+  return obj !== undefined && !uninstallingIDs.has(obj.id);
```

Comment:
IIRC, the correct check should be
```suggestion
  return typeof obj !== 'undefined' && !uninstallingIDs.has(obj.id);
```

Reply:
done

Reply:
That would be the case for JavaScript, and if you expect the variable to be completely undeclared. Here we obviously do have the variable declared and the TypeScript compiler will prevent using undeclared variables anyway. So, `obj !== undefined` is perfectly fine in TypeScript, and preferable since it's shorter/less complex.